kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncThreadPool.java

```diff
@@ -184,29 +202,34 @@ private static class AsyncEventTask<KOut, VOut> implements Runnable {
     private final ProcessingContext originalContext;
     private final AsyncUserProcessorContext<KOut, VOut> wrappingContext;
     private final FinalizingQueue finalizingQueue;
+    private final AsyncProcessorMetricsRecorder metricsRecorder;
 
     private AsyncEventTask(
         final AsyncEvent event,
         final ProcessingContext originalContext,
         final AsyncUserProcessorContext<KOut, VOut> userContext,
-        final FinalizingQueue finalizingQueue
+        final FinalizingQueue finalizingQueue,
+        final AsyncProcessorMetricsRecorder metricsRecorder
     ) {
       this.event = event;
       this.originalContext = originalContext;
       this.wrappingContext = userContext;
       this.finalizingQueue = finalizingQueue;
+      this.metricsRecorder = metricsRecorder;
     }
 
     @Override
     public void run() {
       try {
+        final Instant start = Instant.now();
         wrappingContext.setDelegateForAsyncThread(new AsyncThreadProcessorContext<>(
                 originalContext,
                 event
         ));
         event.transitionToProcessing();
         event.inputRecordProcessor().run();
         finalizingQueue.scheduleForFinalization(event);
+        metricsRecorder.recordEventProcess(Duration.between(start, Instant.now()));
```

Comment:
> another performance issue we encountered in Streams was due to converting between long and Instant/Duration on record fetches.

Interesting. I don't really know anything about the internals with the time api. I figured it was just doing the same math internally. I can switch to manually computed times.

Reply:
re performance of timers: there's going to be _some_ performance hit from reading the time (though it should be a lot less than it used to be now that the default time source on ec2 nodes is sane). My thinking for this scenario is that:
1. these metrics are super useful for debugging, especially in the early phases of this feature.
2. in the context of the use cases we know we want to use this for now, I'm confident the overhead won't be meaningful.
3. the metrics aren't part of a public contract like ak metrics are, so we can change them to debug if/when we feel like we don't want to eat the performance penalty.

Reply:
Yep that all sounds fair. Unless the soak drops like a rock when I deploy with these metrics I won't complain (any more :P)

Reply:
Also it probably all comes back to that same issue with #currentTimeMillis, since `Instant#now` uses that internally. But here we only need #currentTimeNanos which is always fast, since we don't need an exact clock just a relative one. But I'll shut up now